sda-download/internal/config/config.go

```diff
@@ -296,6 +311,31 @@ func (c *Map) configArchive() {
 	}
 }
 
+func (c *Map) configReencrypt() {
+	c.Reencrypt.Host = viper.GetString("grpc.host")
+	viper.SetDefault("grpc.port", 50051)
+	viper.SetDefault("grpc.timeout", 5) // set default to 5 seconds
+	if viper.IsSet("grpc.port") {
+		c.Reencrypt.Port = viper.GetInt("grpc.port")
+	}
+	if viper.IsSet("grpc.timeout") {
+		c.Reencrypt.Timeout = viper.GetInt("grpc.timeout")
+	}
+	if viper.IsSet("grpc.cacert") {
+		c.Reencrypt.CACert = viper.GetString("grpc.cacert")
+		// when certificate is set, the port number should be 50443
+		if c.Reencrypt.Port != 50443 {
+			log.Errorf("CACert is set, but port number is set to %v, which should be 50443", c.Reencrypt.Port)
+		}
```

Comment:
```suggestion
```

The port is not tied to the CA certificate but rather the existence of the client key, see comment below